tests/functional/test_apple_events.py

```diff
@@ -52,3 +53,145 @@ def test_osx_custom_protocol_handler(tmpdir, pyi_builder_spec):
+            else:
+                # Try again later
+                time.sleep(polltime)
+
+    # wait for the app started for us by test_spec to exit
+    assert wait_for_started(), "App did not start"
+
+    time.sleep(2)  # presumably app has exited after 2 seconds
+
+    # clean up the log file created by test_spec() running the app
+    os.remove(logfile_path)
+
+    # Run using 'open', passing a 0-timeout as an arg.
+    # macOS will auto-register the custom protocol handler and extension
+    # association. Then app will quit immediately due to the "0" arg.
+    subprocess.check_call(['open', app_path, '--args', "0"])
+
+    assert wait_for_started(), 'App start timed out'
+    time.sleep(2)  # wait for app to exit
+
+    # App exited immediately, clean-up
+    os.remove(logfile_path)
+
+    # At this point both the protocol handler and the file ext are registered
+    # 1. Try the file extension -- this tests the AppleEvent rewrite of
+    #    a "file://" event to a regular filesystem path.
+
+    # Create 32 files that are associated with this app.
+    # This tests the robustness of the argv-emu by spamming it with
+    # lots of args and seeing what happens.
+    n_files = 32
+    assoc_files = []
+    for ii in range(n_files):
+        assoc_path = os.path.join(tmpdir, 'dist',
+                                  'AFile{}.{}'.format(ii, custom_file_ext))
+        with open(assoc_path, 'wt') as fh:
+            fh.write("File contents #{}\n".format(ii))
+        assoc_files.append(assoc_path)
+
+    # Open app again by "open"ing the associated files.
+    #
+    # These are sent as Apple Events to the app immediately after it starts,
+    # which the bootloader translates back into file paths at startup, passing
+    # them as argv to the subordinate app.
+    #
+    # The generator below produces odd numbered files as "file://" URLs, and
+    # even numbered are just file paths. They all should end up appended to
+    # sys.argv in the app as simple file paths.
+    subprocess.check_call(['open', *[('file://' if ii % 2 else '') + ff
+                                     for ii, ff in enumerate(assoc_files)]])
+
+    args = wait_for_started()
+    assert args is not None, 'App start timed out'
+    # Test that all the file paths were received in argv via pre-startup
+    # translation of file:// AppleEvent -> argv filesystem path.
+    assert assoc_files == args[1:], \
+        "An expected file path was not received by the app"
+
+    # At this point the app is running.
+    # 2. Call open again using the url associated with the app. This should
+    #    forward the Apple URL event to the already-running app.
+    url = custom_url_scheme + "://lowecase_required/hello_world/"
+    # Test support for large URL data ~64KB.
+    # Note: We would have gone larger but 'open' itself seems to not
+    # consistently like data over a certain size.
+    url += 'x' * 64000  # Append 64 KB of data to URL to stress-test
```

Comment:
Strictly speaking 64KB is actually 65536 bytes because there are actually `(1 << 10) == 1024` bytes in a kilobyte. But that doesn't matter - just a fun fact...

Reply:
Speaking of fun facts: 64KB == 64000 bytes.  You must be referring to 64KiB, which is 65536 bytes.  Yes, they renamed it a few years back to be more congruent with the other SI units.  Back when I was a kid, 64KB was in fact 65536.  No more.  Here is more info on this topic:

https://en.wikipedia.org/wiki/Kibibyte